File: csharp/gmsec5/dox/Status.cs
/*
 * Copyright 2007-2022 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */

/* This file is used for generating Doxygen documentation; it only contains
   stubs of constructors and methods associated with the class.
*/


namespace GMSEC.API5 {

/// <summary>
/// This is the class for holding status information returned by the API
/// calls to provide feedback. Status provides an error code and error
/// message specific to the error type, but independent of error source.
/// </summary>

public class Status : global::System.IDisposable {

  /// <summary>Default constructor.</summary>
  public Status() {
  }

  /// <summary>Create a status object with a specific error classification and error code</summary>
  ///
  /// <param name="eclass">Status class</param>
  /// <param name="code">Status code</param>
  public Status(int eclass, int code) {
  }

  /// <summary>Create a status object with a specific error classification, error code, and error reason</summary>
  ///
  /// <param name="eclass">Status class</param>
  /// <param name="code">Status code</param>
  /// <param name="text">Status reason</param>
  public Status(int eclass, int code, string text) {
  }

  /// <summary>Create a status object with a specific error classification, error code, error reason, and custom code</summary>
  ///
  /// <param name="eclass">Status class</param>
  /// <param name="code">Status code</param>
  /// <param name="text">Status reason</param>
  /// <param name="custom">Custom error code</param>
  public Status(int eclass, int code, string text, int custom) {
  }

  /// <summary>Create a status object using info from the given GmsecException object</summary>
  ///
  /// <param name="e">GmsecException object</param>
  public Status(GmsecException e) {
  }

  /// <summary>Copy constructor.</summary>
  public Status(Status other) {
  }

  /// <summary>This will return true if the status is indicating an error, false otherwise</summary>
  public bool HasError() {
  }

  /// <summary>This method will return a verbose status string as follows:
  /// [<i>class</i>,<i>code</i>,<i>custom</i>]: <i>reason</i></summary>
  ///
  /// <returns>Verbose status string</returns>
  public string Get() {
  }

  /// <summary>This is a convenience method that will set class and code.</summary>
  ///
  /// <param name="eclass">Status class</param>
  /// <param name="code">Status code</param>
  public void Set(int eclass, int code) {
  }

  /// <summary>This is a convenience method that will set class, code, and error text.</summary>
  ///
  /// <param name="eclass">Status class</param>
  /// <param name="code">Status code</param>
  /// <param name="text">Status reason</param>
  public void Set(int eclass, int code, string text) {
  }

  /// <summary>This is a convenience method that will set class, code, error text, and a custom error code.</summary>
  ///
  /// <param name="eclass">Status class</param>
  /// <param name="code">Status code</param>
  /// <param name="text">Status reason</param>
  /// <param name="custom">Custom error code</param>
  public void Set(int eclass, int code, string text, int custom) {
  }

  /// <summary>This method will return the status class.</summary>
  ///
  /// <seealso cref="StatusClass" />
  public int GetClass() {
  }

  /// <summary>This will set the status class</summary>
  ///
  /// <param name="eclass">Status class</param>
  public void SetClass(int eclass) {
  }

  /// <summary>This will return the status code.</summary>
  ///
  /// <seealso cref="StatusCode" />
  public int GetCode() {
  }

  /// <summary>This will set the specific status code.</summary>
  ///
  /// <param name="code">status code</param>
  public void SetCode(int code) {
  }

  /// <summary>This method will retrieve the string detailed description.</summary>
  ///
  /// <returns>Detailed Description String</returns>
  public string GetReason() {
  }

  /// <summary>This will set the status description.</summary>
  ///
  /// <param name="reason">reason text</param>
  public void SetReason(string reason) {
  }

  /// <summary>This will return the custom code number for this status.</summary>
  ///
  /// <returns>Custom Code</returns>
  public int GetCustomCode() {
  }

  /// <summary>This will set the custom code.</summary>
  ///
  /// <param name="code">custom code</param>
  public void SetCustomCode(int code) {
  }

  /// <summary>This clears this Status to NO_ERROR.</summary>
  public void Reset() {
  }

}

}

File: nodejs/gmsec5/dox/Status.java
/*
 * Copyright 2007-2024 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */


/**
 * @file Status.java
 */

package gmsec;


/**
 * This is the class for holding status information returned by the API
 * function calls to provide feedback. Status provides an error code
 * and error message specific to the error type, but independent of error
 * source.
 */
public class Status
{
	/**
	 * Constructor.
	 */
	public Status();


	/**
	 * Constructs a Status object with the given error class, code and message.
	 *
	 * @param clazz  Integer value indicating source of the error
	 * @param code   Integer value indicating reason for the error
	 * @param reason String providing explanation for the error
	 */
	public Status(clazz, code, reason);


	/**
	 * Constructs a Status object with the given error class, code, custom code and message.
	 *
	 * @param clazz      Integer value indicating source of the error
	 * @param code       Integer value indicating reason for the error
	 * @param reason     String providing explanation for the error
	 * @param customCode Custom integer error code (typically middleware specific) regarding the root cause of the error
	 */
	public Status(clazz, code, reason, customCode);


	/**
	 * Construct Status object using values from an existing Status object.
	 *
	 * @param other Status object to copy.
	 *
	 * @throws SWIG_Exception Thrown if the given Status object is null.
	 */
	public Status(other);


	/**
	 * Used to determine whether the Status object is reporting an error or not.
	 * <p>
	 * Note: Only the status class is examine; the status code is not checked.
	 *
	 * @return False if the status class is set to NO_ERROR; true otherwise.
	 */
	public hasError();


	/**
	 * This function will return a verbose error string that contains the
	 * Status class, code, custom code and reason.
	 * <p>
	 * The format is as follows:
	 * <pre>{@code
	 * [class,code,custom code] : reason
	 * }</pre>
	 *
	 * @return The error string associated with the Status object.
	 */
	public get();


	/**
	 * Used to set error class, error code, and error text for the Status object.
	 *
	 * @param clazz  Integer value indicating source of the error
	 * @param code   Integer value indicating reason for the error
	 * @param reason String providing explanation for the error
	 */
	public set(clazz, code, reason);


	/**
	 * Used to set error class, error code, custom code, and error text for the Status object.
	 *
	 * @param clazz      Integer value indicating source of the error
	 * @param code       Integer value indicating reason for the error
	 * @param reason     String providing explanation for the error
	 * @param customCode Custom integer error code (typically middleware specific) regarding the root cause of the error
	 */
	public set(clazz, code, reason, customCode);


	/**
	 * Returns the error class associated with the Status object.
	 *
	 * @return A error class value as an integer.
	 */
	public getClassification();


	/**
	 * This will set the status classification.
	 *
	 * @param clazz  Integer value indicating source of the error
	 */
	public setClassification(clazz);


	/**
	 * Returns the status code associated with the Status object.
	 *
	 * @return A status code value as an integer.
	 */
	public getCode();


	/**
	 * This will set the status code.
	 *
	 * @param code Integer value indicating reason for the error
	 */
	public setCode(code);


	/**
	 * Returns the error message associated with the Status object.
	 *
	 * @return A string containing the error message.
	 */
	public getReason();


	/**
	 * This will set the error string text.
	 *
	 * @param reason String providing explanation for the error
	 */
	public setReason(reason);


	/**
	 * Returns the custom code associated with the Status object.
	 *
	 * @return A custom code value as an integer.
	 */
	public getCustomCode();


	/**
	 * This will set the (middleware) specific error code.
	 *
	 * @param customCode Integer value for custom code, typically middleware specific.
	 */
	public setCustomCode(customCode);


	/**
	 * This clears the Status object to a 'no error' state.
	 */
	public reset();
}
